### ISSUE TITLE
Add an optional focus ring to `VLink` component

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse-frontend#2049]
## Description
Some links require focus ring. To ensure that the focus rings are always consistent, we should add a `focus-ring` prop to `VLink`, and add the necessary tailwind classes to add a focus ring if the prop is true. By default, the prop should be false.

## Note
Based on the discussion in #2015: 
    > I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages

Very much agree with this. My point was about the need for `8px` of padding as the focus style looks strange.

<img width="353" alt="CleanShot 2022-12-07 at 19 16 29@2x" src="https://user-images.githubusercontent.com/895819/206263469-26ef19ce-13bf-41ec-a1e7-8b1ecee426d8.png">

Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

What you did is definitely correct. I am just bringing this up to reach consistency across other similar elements. I am thinking of the "Clear" action in filters, "Report this content" in the single result, these nav links in the Footer, and possibly others that I am missing.

The `Clear` component does exist in Figma, but since it does not in code, and we are discussing its purpose, it seems timely to define a basic link element that does not have paddings/margins, could have an icon, and where focus ring wraps the element. Something like the image below, but called `VSingleLink` instead (name TBD).

<img width="300" alt="CleanShot 2022-12-07 at 19 24 54@2x" src="https://user-images.githubusercontent.com/895819/206265159-76b8e9eb-a6e3-4993-8948-dd59235c5708.png">

What do you think?

_Originally posted by @panchovm in https://github.com/WordPress/openverse-frontend/issues/2015#issuecomment-1341424072_

**Comment by panchovm** [comment]
The ring styles we are using have two thicknesses for two cases. Here is a table I made.

<img width="1347" alt="CleanShot 2022-12-19 at 19 05 46@2x" src="https://user-images.githubusercontent.com/895819/208491156-84a09763-f45f-443c-bbc1-b6d75f018980.png">

It would be great to add these classes and reach more consistency. I still notice some different focus ring styles when testing components.

**Comment by dhruvkb** [comment]
@panchovm this is a very helpful table! See all the different focus ring variations at once will be useful to define the Tailwind config accordingly.

**Comment by panchovm** [comment]
I noticed from PR #2060 implementation that there might be a misunderstanding about what [this diagram](https://github.com/WordPress/openverse-frontend/issues/2049#issuecomment-1358044819) defines as outside and inside.

* The **Inside** style is a border that inwardly adds stroke weight, masking the content. 
* The **Outside** style is a border that outwardly adds stroke weight.

Here are some screenshots from Figma mockups where the red lines mark the component area. Notice how borders go inward and outward.

### `Slim transparent`

|Resting|Focus|
|:---|:---|
|<img width="150" alt="CleanShot 2022-12-26 at 16 04 50@2x" src="https://user-images.githubusercontent.com/895819/209561879-84667d43-5cf2-48f0-b03a-79b4a1d2ebe8.png">|<img width="150" alt="CleanShot 2022-12-26 at 16 05 13@2x" src="https://user-images.githubusercontent.com/895819/209561905-5e568a97-906f-44c2-aec8-b8f15d205422.png">|

### `Slim filled`

|Resting|Focus|
|:---|:---|
|<img width="150" alt="CleanShot 2022-12-26 at 16 07 46@2x" src="https://user-images.githubusercontent.com/895819/209562127-9d0e8a4c-6e05-4326-98d5-ca7f269532e0.png">|<img width="150" alt="CleanShot 2022-12-26 at 16 08 53@2x" src="https://user-images.githubusercontent.com/895819/209562200-2e0d2fe6-001c-46ec-84bf-2ef373fa0652.png">|

### `Bold filled`

|Resting|Focus|
|:---|:---|
|<img width="150" alt="CleanShot 2022-12-26 at 16 06 50@2x" src="https://user-images.githubusercontent.com/895819/209562044-0d7cf2f6-d736-42a1-81ec-b1af363896a9.png">|<img width="150" alt="CleanShot 2022-12-26 at 16 06 54@2x" src="https://user-images.githubusercontent.com/895819/209562047-460bb20e-3a14-4aa6-9ea5-8cd6f647022d.png">|

**Comment by dhruvkb** [comment]
@panchovm #2067 adds new utilities for focus, can you check if that matches your expected behaviour?

**Comment by panchovm** [comment]
Based on a synchronized conversation I had with @dhruvkb, [this diagram](https://github.com/WordPress/openverse-frontend/issues/2049#issuecomment-1358044819) lacks some information, so here is an updated description.

Every focus style has `2px` for border-radius and mixes one value of each variable.

## Variables

<img src="https://user-images.githubusercontent.com/895819/209824898-8da4733c-cb8e-4e54-8f07-f813bd2ac1cc.png" width="600">

#### Note: The `bold` + `transparent` combination does not exist so far.

## Examples
<img src="https://user-images.githubusercontent.com/895819/209826379-25ca6106-a61a-471f-b74a-9444d3739a14.png" width="382">

<img src="https://user-images.githubusercontent.com/895819/209826388-14fac114-6b85-4358-9ddf-27fcfcbd9eee.png" width="439">

<img src="https://user-images.githubusercontent.com/895819/209826396-e29a70d7-e9b7-4616-ad5d-729fb45e8278.png" width="340">

**Comment by dhruvkb** [comment]
@panchovm and I are having a discussion in Figma regarding links in the mobile navigation screen. I will implement a PR for this when that discussion is resolved.
